File: algorithms/strStr/strStr.go
package main

import "fmt"

func main() {
	fmt.Println(strStr("hello", "ll"))
	fmt.Println(strStr("aaaaa", "bba"))
	fmt.Println(strStr("", ""))
}

func strStr(haystack string, needle string) int {
	// special case
	if needle == "" {
		return 0
	}

	for hidx := range haystack {
		matched := 0
		for range needle {
			hloc := hidx + matched
			if hloc < len(haystack) && haystack[hloc] == needle[matched] {
				matched++
			} else {
				break
			}

			if matched == len(needle) {
				return hidx
			}
		}
	}

	return -1
}

File: str_str.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        return haystack.find(needle)


s = Solution()
print(s.strStr("hello", "ll"))
print(s.strStr("aaaaa", "bba"))
print(s.strStr("", ""))
